src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -78,13 +95,78 @@ public ApplicationInsightsLogger(TelemetryClient client, string categoryName)
             }
         }
 
+        private void LogMetric(IEnumerable<KeyValuePair<string, object>> values)
+        {
+            MetricTelemetry telemetry = new MetricTelemetry();            
+
+            foreach (var entry in values)
+            {
+                if (entry.Value == null)
+                {
+                    continue;
+                }
+
+                // Name and Value are not lower-case so check for them explicitly first and move to the
+                // next entry if found.
+                switch (entry.Key)
+                {
+                    case LogConstants.NameKey:
+                        telemetry.Name = entry.Value.ToString();
+                        continue;
+                    case LogConstants.MetricValueKey:
```

Comment:
That really wouldn't be usable with `LogMetric()` as you'd have to ignore the `value` parameter and add a `sum` entry to the properties. And we'd need to say `sum` wins if there's a conflict between `sum` and `value`. It feels more confusing that way. I will have to document the App Insights behavior -- I'll do that in the wiki once this is all in.